src/client/serialize/daos_serialize.c

```diff
@@ -0,0 +1,1026 @@
+		D_GOTO(out, rc);
+	}
+	attr_dspace = H5Aget_space(cont_attr);
+	if (status < 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to read acl dspace "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	ndims = H5Sget_simple_extent_dims(attr_dspace, attr_dims, NULL);
+	if (ndims < 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to get dimensions of dspace "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	D_ALLOC(rdata, attr_dims[0] * sizeof(char *));
+	if (rdata == NULL) {
+		rc = -DER_NOMEM;
+		D_GOTO(out, rc);
+	}
+	attr_dtype = H5Tcopy(H5T_C_S1);
+	if (status < 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to create dtype "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	status = H5Tset_size(attr_dtype, H5T_VARIABLE);
+	if (status < 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to set acl dtype size "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	status = H5Aread(cont_attr, attr_dtype, rdata);
+	if (status < 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to read property attribute "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	/* convert acl strings back to struct acl, then store in entry */
+	rc = daos_acl_from_strs(rdata, (size_t)attr_dims[0], &acl);
+	if (rc != 0) {
+		D_ERROR("failed to convert acl strs "DF_RC"\n", DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	entry->dpe_val_ptr = (void *)acl;
+out:
+	D_FREE(rdata);
+
+	/* close hdf5 objects */
+	if (cont_attr > 0)
+		H5Aclose(cont_attr);
+	if (attr_dtype > 0)
+		H5Tclose(attr_dtype);
+	if (attr_dspace > 0)
+		H5Sclose(attr_dspace);
+	return rc;
+}
+
+static int
+deserialize_props(daos_handle_t poh, hid_t file_id, daos_prop_t **_prop,
+		  uint64_t *cont_type)
+{
+
+	int			rc = 0;
+	bool			deserialize_label = false;
+	bool			close_cont = true;
+	uint32_t		num_props = 16;
+	daos_prop_t		*label = NULL;
+	daos_prop_t		*prop = NULL;
+	struct daos_prop_entry	*entry;
+	struct daos_prop_entry	*label_entry;
+	daos_handle_t		coh;
+	daos_cont_info_t	cont_info = {0};
+
+	label = daos_prop_alloc(1);
+	if (label == NULL) {
+		return ENOMEM;
+	}
+	label->dpp_entries[0].dpe_type = DAOS_PROP_CO_LABEL;
+
+	/* read the container label entry to decide if it should be added
+	 * to property list. The container label is required to be unique,
+	 * which is why it is handled differently than the other container
+	 * properties. If the label already exists in the pool then this
+	 * property will be skipped for deserialization
+	 */
+	label_entry = &label->dpp_entries[0];
+	rc = deserialize_str(file_id, label_entry, "DAOS_PROP_CO_LABEL");
+	if (rc != 0) {
+		D_GOTO(out, rc);
+	}
+	rc = daos_cont_open_by_label(poh, label_entry->dpe_str, DAOS_COO_RW,
+				     &coh, &cont_info, NULL);
+	if (rc == -DER_NONEXIST) {
+		/* doesn't exist so ok to deserialize this container label */
+		deserialize_label = true;
+		close_cont = false;
+	} else if (rc != 0) {
+		D_GOTO(out, rc);
+	}  else {
+		fprintf(stdout, "Container label cannot be set, "
+			"the label already exists in pool\n");
+	}
+
+	if (deserialize_label) {
+		num_props++;
+	}
```

Comment:
I think it's just more of a preference, but unlike mpifileutils, the daos repo is okay with this
```suggestion
	if (deserialize_label)
		num_props++;
```